app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryEffect.kt

```diff
@@ -1,3 +1,7 @@
 package org.simple.clinic.drugs.selection.entry
 
+import java.util.UUID
+
 sealed class CustomPrescriptionEntryEffect
+
+data class SaveCustomPrescription(val patientUuid: UUID, val drugName: String, val dosage: String?) : CustomPrescriptionEntryEffect()
```

Comment:
iirc, `Optional` is for cases where are using the value in `Rx` streams and `Nullable` is for immediate use.

Reply:
Right, but that's because Rx lambdas will throw exceptions if an operator returns `null`. I'm thinking if we need to use the `Optional` type itself to expose some intent here.

Reply:
Since `dosage` in the model is also nullable and not `Optional`, I'd much rather keep it this way for now.

Reply:
Yeah, this was just a wandering thought.